DataDog/datadog-operator
# Migrate to kubebuilder v4

### What does this PR do?

Migrate project to use kubebuilder v4: https://book.kubebuilder.io/migration/manually_migration_guide_gov3_to_gov4

Main changes: 
- `controllers/` renamed and moved to `internal/controller/`
- `apis/` renamed to `api/`
- `main.go` moved to `cmd/main.go`
- kubebuilder dependency version bumps
  - kubebuilder v4.1.1
  - kustomize/v5@v5.4.3
  - golangci-lint v1.59.1
  


### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

* Run `make all` and  there should be no errors and tests pass
* Run `make VERSION=X.Y.Z-rc.W LATEST_VERSION=X.Y.Z bundle` and check that bundle is created properly and does not produce errors in the terminal
* Same with `make redhat-bundle`
* Run `make install` in a fresh kind cluster and CRDs should install 
* Run `make deploy` and check that the operator pod is running properly
* Deploy a datadogagent manifest and spot check that agent pods get created properly

### Checklist
